src/components/modals/TextInputModal.tsx

```diff
@@ -99,57 +99,56 @@ export function TextInputModal(props: Props) {
-        {warningMessage == null ? null : (
-          <Alert
-            type="warning"
-            title={s.strings.warning}
-            marginRem={0.5}
-            message={warningMessage}
-            numberOfLines={0}
-          />
-        )}
-        <OutlinedTextInput
-          // Text input props:
-          autoCapitalize={autoCapitalize}
-          autoFocus={autoFocus}
-          autoCorrect={autoCorrect}
-          keyboardType={keyboardType}
-          label={inputLabel}
-          returnKeyType={returnKeyType}
-          secureTextEntry={secureTextEntry}
-          multiline={multiline}
-          // Our props:
-          error={errorMessage}
-          marginRem={[1, 0.5, 1.5, 0.5]}
-          onChangeText={handleChangeText}
-          onSubmitEditing={handleSubmit}
-          value={text}
-          maxLength={maxLength}
+      {message == null ? null : <ModalMessage>{message}</ModalMessage>}
+      {warningMessage == null ? null : (
+        <Alert
+          type="warning"
+          title={s.strings.warning}
+          marginRem={0.5}
+          message={warningMessage}
+          numberOfLines={0}
         />
-        {
-          // Hack around the android:windowSoftInputMode="adjustPan" glitch:
-          Platform.OS === 'android' ? <View style={{ flex: 2 }} /> : null
-        }
-        {spinning ? (
-          <MainButton
-            alignSelf="center"
-            disabled
-            marginRem={0.5}
-            type="secondary"
-            spinner
-          />
-        ) : (
-          <MainButton
-            alignSelf="center"
-            label={submitLabel}
-            marginRem={0.5}
-            onPress={handleSubmit}
-            type="secondary"
-          />
-        )}
-      </ModalScrollArea>
+      )}
+      <OutlinedTextInput
+        // Text input props:
+        autoCapitalize={autoCapitalize}
+        autoFocus={autoFocus}
+        autoCorrect={autoCorrect}
+        keyboardType={keyboardType}
+        label={inputLabel}
+        returnKeyType={returnKeyType}
+        secureTextEntry={secureTextEntry}
+        multiline={multiline}
+        // Our props:
+        error={errorMessage}
+        marginRem={[1, 0.5, 1.5, 0.5]}
+        onChangeText={handleChangeText}
+        onSubmitEditing={handleSubmit}
+        value={text}
+        maxLength={maxLength}
+      />
+      {
+        // Hack around the android:windowSoftInputMode="adjustPan" glitch:
+        Platform.OS === 'android' ? <View style={{ flex: 2 }} /> : null
+      }
+      {spinning ? (
+        <MainButton
+          alignSelf="center"
+          disabled
+          marginRem={[0.5, 0.5, 5, 0.5]}
+          type="secondary"
+          spinner
+        />
+      ) : (
+        <MainButton
+          alignSelf="center"
+          label={submitLabel}
+          marginRem={[0.5, 0.5, 5, 0.5]}
+          onPress={handleSubmit}
+          type="secondary"
+        />
+      )}
+      <ModalFooter onPress={handleCancel} fadeOut />
```

Comment:
Delete the `fadeOut` part, and then the `marginRem` props can go back to being 0.5 all around.